docs/device_resources.md

```diff
@@ -0,0 +1,224 @@
+## Device Resources
+
+The device resources schema is intended to be a complete description of an
+island-based FPGA design.  It is made of many components, but the core
+description of the device is shown below.
+
+```
+┌─────────────────┐
+│  Device         │
+│ ┌─────────────┐ │
+│ │ Tile        │ │
+│ │ ┌─────────┐ │ │
+│ │ │ Site    │ │ │
+│ │ │ ┌─────┐ │ │ │
+│ │ │ │ BEL │ │ │ │
+│ │ │ └─────┘ │ │ │
+│ │ │         │ │ │
+│ │ └─────────┘ │ │
+│ │             │ │
+│ └─────────────┘ │
+│                 │
+└─────────────────┘
+```
+
+That is:
+ - A device contains tiles
+ - Tiles contains sites
+ - Sites contains BELs
+
+The schema contains the required information to answer questions such as:
+ - Where are tiles are located?
+ - How are sites connected to the routing graph?
+ - How are BELs connected to the boundry of the site?
+ - How can cells be placed at BELs?
+
+## Terminology
+
+- Device - A set of tiles and package pins.
+- Tiles - An instance of a tile type which contains wires and sites
+- Package pin - A boundry between the "interior" of the device and what is "outside" the package.  Generally corrisponds to a pin on a package, e.g. pin 1 on SOP-8 or A1 on CSG324.
+- Wire - Also known as a "tile wire" .  A wire is a piece of conductive material totally contained within a tile.  Wires can be part of nodes.  Wires can connect to PIPs or site pins.
+- Node - A node is a set of 1 or more wires that are connected.  Nodes can span multiple tiles.  Nodes connect to PIPs or site pins via the wires that are part of the node.
+- PIP - PIP is an abreviation for programable interconnect point.  A PIP provides a connection between two wires.  PIPs can be unidirectional or bidirectional.  Unidirectional PIPs always connect wire0 to wire1.  Bidirectional PIPs can connect wire0 to wire1 or wire1 to wire0.
+- Site - A collection of site pins, site wires and BELs.
```

Comment:
I think an example of a site (a SLICE) could be useful here

Reply:
Done